routes/events/events.js

```diff
@@ -8,21 +9,61 @@ const eventRouter = express();
 // get all events
 eventRouter.get('/', async (req, res) => {
   try {
-    const conditions = '';
-    const events = await pool.query(getEventsQuery(conditions));
-    res.status(200).json(keysToCamel(events.rows));
-  } catch (err) {
-    res.status(400).send(err.message);
-  }
-});
+    console.log(req.query);
+    const { status, type } = req.query;
+    // const offset = (pageIndex - 1) * pageSize;
+    const search = req.query.search == null ? '' : req.query.search;
+    const timeComparisonDict = {
+      upcoming: `start_datetime >= NOW()`,
+      past: `start_datetime < NOW()`,
+      all: `-1 = -1`,
+    };
+    const typeDict = {
+      'food running': `event_type = 'Food Running'`,
+      distribution: `event_type = 'Distribution'`,
+      other: `(event_type != 'Food Running' AND event_type != 'Distribution')`,
+      all: `-1 = -1`,
+    };
+    const eventFilter = `
+    SELECT *
+    FROM events
+    LEFT JOIN
+      (SELECT req.event_id AS eid, array_agg(req.requirement ORDER BY req.requirement ASC) AS requirements
+        FROM event_requirements AS req
+        GROUP BY req.event_id) AS r on r.eid = events.event_id
+    LEFT JOIN
+      (SELECT waivers.event_id AS eid, array_agg(to_jsonb(waivers.*) - 'event_id' ORDER BY waivers.name) AS waivers
+        FROM waivers
+        GROUP BY waivers.event_id) AS waivers on waivers.eid = events.event_id
+    WHERE (${timeComparisonDict[status]})
+      AND (${typeDict[type]})
+    `;
+    let events = await db.query(eventFilter);
```

Comment:
let's use `const` for this `events` variable and in the following if loop just `res.status(200).json( keysToCamel the filtered list )` instead of changing the variable to return + add an `else { res.status(200).json( keysToCamel events ) }` , in general `let` isn't really a good thing to use unless absolutely necessary for maintainability reasons